packages/dashboard/src/components/Restore/RestoreFromCloud.tsx

```diff
@@ -38,30 +54,59 @@ export const RestoreFromCloud = memo(() => {
         enqueueSnackbar(t.sign_in_account_cloud_backup_download_failed(), { variant: 'error' })
     }, [fetchBackupValueError])
 
-    const onValidated = async (downloadLink: string, account: string, password: string) => {
-        try {
+    const onValidated = useCallback(
+        async (downloadLink: string, accountValue: string, password: string, type: AccountType) => {
             const backupValue = await fetchBackupValueFn(downloadLink)
-            const backupText = await decryptBackupFn(account, password, backupValue)
-            const backupInfo = await Services.Welcome.parseBackupStr(backupText)
+            const backupText = await decryptBackupFn(accountValue, password, backupValue)
 
+            if (!backupText) {
+                return t.sign_in_account_cloud_backup_decrypt_failed()
+            }
+
+            const backupInfo = await Services.Welcome.parseBackupStr(backupText)
             if (backupInfo) {
                 setBackupId(backupInfo.id)
-                setStep({ name: 'restore', params: { backupJson: backupInfo.info } })
+                setAccount({ type, value: accountValue, password })
+                setStep({
+                    name: 'restore',
+                    params: {
+                        backupJson: backupInfo.info,
+                        handleRestore: () => onRestore(backupInfo.id, { type, value: accountValue, password }),
+                    },
+                })
+                return null
             }
-            return null
-        } catch {
             return t.sign_in_account_cloud_backup_decrypt_failed()
-        }
-    }
+        },
+        [],
+    )
 
-    const onRestore = async () => {
-        try {
-            await Services.Welcome.checkPermissionsAndRestore(backupId)
-            navigate(RoutePaths.Personas, { replace: true })
-        } catch {
-            enqueueSnackbar(t.sign_in_account_cloud_restore_failed(), { variant: 'error' })
-        }
-    }
+    const onRestore = useCallback(
+        async (backupId: string, account: any) => {
+            try {
+                await Services.Welcome.checkPermissionsAndRestore(backupId)
+
+                if (!currentPersona) {
+                    const lastedPersona = await Services.Identity.queryLastPersonaCreated()
+                    if (lastedPersona) {
+                        await changeCurrentPersona(lastedPersona.identifier)
+                    }
+                }
+                if (account) {
+                    if (!user.email && account.type === AccountType.email) {
+                        updateUser({ email: account.value })
+                    }
+                    if (!user.phone && account.type === AccountType.phone) {
+                        updateUser({ phone: account.value })
```

Comment:
phone number: ``${countryCode} ${phone}``

Reply:
@yanzhihong23 plz read E.164 standard

Reply:
@septs not talking standard here. This value will be saved in localStorage. Just for separating country code and number.